admission/webhook/validator.go

```diff
@@ -9,26 +9,39 @@ import (
 	"github.com/kubescape/k8s-interface/k8sinterface"
 	exporters "github.com/kubescape/operator/admission/exporter"
 	"github.com/kubescape/operator/admission/rulebinding"
+	"github.com/kubescape/operator/objectcache"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/client-go/kubernetes"
 )
 
 type AdmissionValidator struct {
 	kubernetesClient *k8sinterface.KubernetesApi
+	objectCache      objectcache.ObjectCache
 	exporter         *exporters.HTTPExporter
 	ruleBindingCache rulebinding.RuleBindingCache
 }
 
 func NewAdmissionValidator(kubernetesClient *k8sinterface.KubernetesApi, exporter *exporters.HTTPExporter, ruleBindingCache rulebinding.RuleBindingCache) *AdmissionValidator {
+	kubernetesCache := objectcache.NewKubernetesCache(kubernetesClient)
```

Comment:
You should use dependency injection, meaning, pass the objectcache interface to the admission validator instead of initializing it inside.